### PR TITLE
Move to GitHub Actions

### PR DESCRIPTION
**Pull request by juanpicado**
Ref https://github.com/Level/community/issues/99

Currently are failing due `Sauce Labs credentials are required` error on my repo.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +0,0 @@
-language: node_js
-
-node_js: stable
-
-script:
-  - npm run test-browsers
-
-addons:
-  hosts:
-    - airtap.local
```
Comment (juanpicado) [inline .travis.yml:10]: not sure what this does

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [12, 14]
```
Comment (juanpicado) [inline .github/workflows/test.yml:8]: the original is `stable`, we want only one here? 12? 14?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use node ${{ matrix.node }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install
+        run: npm install
+      - name: Test
+        run: npm run test-browsers
```
Comment (juanpicado) [inline .github/workflows/test.yml:20]: there is a single `npm test` at the repo, not sure which one is wished here.

Comment (vweevers) [reply .github/workflows/test.yml:20]: You picked the right one! The `test-browsers` script is for CI while `test` is for local use.

File: .github/workflows/test.yml (hunk 2 above)
Comment (vweevers) [reply .github/workflows/test.yml:8]: Only 14 is fine. The node version doesn't really matter, the only concern is that `airtap` (which runs the browser tests) supports that node version.

File: .travis.yml (hunk 1 above)
Comment (vweevers) [reply .travis.yml:10]: It adds the `airtap.local` hostname to `/etc/hosts`. To do the same in GitHub Actions and also add the required secrets, you can copy https://github.com/nodejs/readable-stream/blob/bcbe5e06b10a8915f622e6530f057f7a4ee1a09e/.github/workflows/sauce.yml#L13-L24

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Test
+on: [push, pull_request]
```
Comment (vweevers) [inline .github/workflows/test.yml:2]:
```suggestion
on: push
```

Because secrets are not available to pull requests from a fork.

**Comment by vweevers** [comment]
We're almost there 😃 To make coverage reports work, can you update the `coverage` npm script and remove `coveralls` from devDependencies? See the similar change to `package.json` in https://github.com/Level/concat-iterator/commit/a229bb4c8ae407debe0ea2bbed5f2d0a6082b988#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

File: package.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
   "scripts": {
     "test": "standard && hallmark && airtap -p local --coverage test/index.js",
     "test-browsers": "standard && airtap --coverage test/index.js",
-    "coverage": "nyc report --reporter=text-lcov | coveralls",
+    "coverage": "nyc report -r lcovonly",
```
[the remark this answers is not in the document]
Comment (vweevers) [reply package.json:11]: It's called here:

https://github.com/Level/level-js/blob/ed57a99ff80632bb4469d8ab2ed42c2393e74502/.github/workflows/test.yml#L29-L30

After the tests finish, we have raw code coverage in `.nyc_output/`. Then `npm run coverage` converts it to LCOV which we then upload to Codecov.

**Review by vweevers — APPROVED** [review]
Amazing, thank you! I will shortly create a branch (that has access to secrets) so we can see it in action (pun intended).

**Comment by juanpicado** [comment]
> Amazing, thank you! I will shortly create a branch (that has access to secrets) so we can see it in action (pun intended).

super thanks :) I'll back tomorrow to convert one more.

**Comment by vweevers** [comment]
Tests passed and coverage works: https://app.codecov.io/gh/Level/level-js

**Pull request merged**